server/scripts/fhirDefinitions.ts

```diff
@@ -0,0 +1,77 @@
+import * as fs from 'fs'
+import { join } from 'path'
+
+import { Photon, StructureDefinitionCreateInput } from '@prisma/photon'
+
+import { HL7_AUTHOR } from '../src/constants'
+
+const photon = new Photon()
+
+const parseBundles = async (directory: string) => {
+  const dir = await fs.promises.opendir(directory)
+  for await (const dirent of dir) {
+    const rawBundle = await fs.promises.readFile(join(dir.path, dirent.name), {
+      encoding: 'utf-8',
+    })
+    const bundle = JSON.parse(rawBundle)
+    if (!bundle.resourceType || bundle.resourceType !== 'Bundle') {
+      throw new Error(`${dirent.name} must be a FHIR Bundle resource`)
+    }
+    await Promise.all(
+      bundle.entry.map(async ({ resource }: any) => {
+        if (!resource) throw new Error('Bundle entry is missing a resource')
+        if (resource.resourceType !== 'StructureDefinition') return
+
+        const definition: StructureDefinitionCreateInput = {
+          id: resource.id,
+          name: resource.name,
+          type: resource.type,
+          description: resource.description,
+          kind: resource.kind,
+          derivation: resource.derivation,
+          author: HL7_AUTHOR,
+          content: JSON.stringify(resource),
+        }
+        await photon.structureDefinitions.create({ data: definition })
+      }),
+    )
+  }
+}
+
+const countDefinitions = async () => {
+  // resources
+  const resources = await photon.structureDefinitions.findMany({
+    where: { derivation: 'specialization', kind: 'resource' },
+    select: { id: true },
```

Comment:
Just out of curiosity: do we need to have the select field not empty?

Reply:
we don't "need" it, but this query just counts the number of records in the DB so I don't want to overfetch (if select is not empty, all fields are fetched)

Reply:
You mean "if select **is** empty, all fields are fetched"?
Thanks, this answers my question 👍